openeogeotrellis/geopysparkdatacube.py

```diff
@@ -312,6 +312,36 @@ def add_dimension(self, name: str, label: str, type: str = None):
             metadata=self.metadata.add_dimension(name=name, label=label, type=type)
         )
 
+    def drop_dimension(self, dimension: str = "other"):
+        if dimension not in ['bands', 'temporal']:
+            raise OpenEOApiException(status_code=400, message=
+                """'drop_dimension' is only supported for dimension types 'bands' and 'temporal'.""")
+
+        if dimension == 'temporal':
+            pyramid = Pyramid(map(lambda l: l.to_spatial_layer(), self.pyramid.levels))
+            return GeopysparkDataCube(
+                pyramid=pyramid,
+                metadata=self.metadata.drop_dimension(type=dimension)
+            )
+        else:
+            if not len(self.metadata.bands) == 1:
+                raise OpenEOApiException(status_code=400, message=
+                    """Band dimension can only be dropped if there is only 1 band left in the datacube""")
```

Comment:
exception should have `code="DimensionLabelCountMismatch"` 

see https://processes.openeo.org/#drop_dimension